### PR TITLE
Add paths elements to classpath

### PR DESCRIPTION
**Pull request by bolivier**
Looks like `path/resolve` is already used for filesystem interactions in main.cljs.  I tested this locally with `nbb -b core` as described in #254 

- [x] I have read the [developer documentation](https://github.com/babashka/nbb/blob/main/doc/dev.md).

- [x] This PR corresponds to an [issue with a clear problem statement](https://github.com/babashka/nbb/blob/main/doc/dev.md#start-with-an-issue-before-writing-code).

#254

- [x] This PR contains a [test](https://github.com/babashka/nbb/blob/main/doc/dev.md#tests) to prevent against future regressions

I'm not sure the best way to test this.  I poked around for similar test cases and couldn't find clarification. 

- [x] I have updated the [CHANGELOG.md](https://github.com/babashka/nbb/blob/main/CHANGELOG.md) file with a description of the addressed issue.

### CODE REVIEW

File: src/nbb/impl/main.cljs [hunk 1]
```diff
@@ -132,6 +132,10 @@ Tooling:
         _ (do (cp/add-classpath cwd)
               (when classpath
                 (cp/add-classpath classpath)))
+        _ (let [nbb-edn (local-nbb-edn)]
+            (doseq [path (:paths nbb-edn)]
+              (cp/add-classpath
+               (path/resolve path))))
```
Comment (borkdude) [inline src/nbb/impl/main.cljs:138]: I don't think calling path/resolve is necessary here.

**Comment by borkdude** [comment]
As for testing, I think it might be good to add a test project in `test-scripts` and then writing an integration test for it in `script/nbb_test.clj`: these tests are integration tests ran with bb.

**Comment by bolivier** [comment]
@borkdude Should the project dir still be added to the classpath if you specify `:paths`?  

My personal preference would be no, because it seems more consistent with the rest of the Clojure ecosystem, but that introduces potentially annoying specificity.  I'm not sure if there's a use case for wanting the project dir _not_ to be in the classpath, anyway.

**Comment by borkdude** [comment]
@bolivier I think you're right about that

File: script/nbb_tests.clj [hunk 2]
```diff
@@ -98,13 +98,19 @@
...
+  (testing "passing classpath cli arg"
+    (let [deps '{com.github.seancorfield/honeysql {:git/tag "v2.0.0-rc5" :git/sha "01c3a55"}}
+          _ (deps/add-deps {:deps deps})
+          cp (cp/get-classpath)]
+      (is (= ["SELECT foo FROM bar WHERE baz = ?" 2]
+             (nbb "--classpath" cp
+                  "-e"
+                  "(require '[honey.sql :as sql]) (sql/format {:select :foo :from :bar :where [:= :baz 2]})")))))
+  (testing "add `:paths` from nbb.edn to classpath"
+    (nbb {:dir "test-scripts/paths-test"} "runner.cljs"))
```
Comment (borkdude) [inline script/nbb_tests.clj:110]: There is a missing test assertion here

Comment (bolivier) [reply script/nbb_tests.clj:110]: @borkdude I'm not sure what I'd assert. The test fails (throws) before adding the new code in `main.cljs`.

Comment (borkdude) [reply script/nbb_tests.clj:110]: Just wrap `(is ...)` around it.

File: script/nbb_tests.clj [hunk 3]
```diff
@@ -107,7 +107,7 @@
                   "-e"
                   "(require '[honey.sql :as sql]) (sql/format {:select :foo :from :bar :where [:= :baz 2]})")))))
   (testing "add `:paths` from nbb.edn to classpath"
-    (nbb {:dir "test-scripts/paths-test"} "runner.cljs"))
+    (is (nbb {:dir "test-scripts/paths-test"} "runner.cljs")))
```
Comment (borkdude) [inline script/nbb_tests.clj:110]: What is the actual value returned here? A string right? Could we do an assertion on that?

Comment (bolivier) [reply script/nbb_tests.clj:110]: Right now it's returning `nil` but I should be able to modify the tested function to return a string

**Pull request merged**